From 9b58fdf0cd3c11ad39387416aff49afc713401af Mon Sep 17 00:00:00 2001
From: Ralf Baechle <rbaechle@caviumnetworks.com>
Date: Wed, 9 Jan 2013 19:42:19 +0100
Subject: [PATCH 301/337] MIPS: Octeon: Fix possible hang to to erroneous
 reuse of loop variable.

Based On SDK 3.0.0-482

Two nested loops should better not use the loop variable.  This may have
resulted in the outer loop looping forever however for some reason this
has only been observed when the kernel was launched by kexec, never
when booting from U-Boot.

Signed-off-by: Ralf Baechle <rbaechle@caviumnetworks.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
Signed-off-by: Jack Tan <jack.tan@windriver.com>
---
 arch/mips/cavium-octeon/setup.c |   11 ++++++-----
 1 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/arch/mips/cavium-octeon/setup.c b/arch/mips/cavium-octeon/setup.c
index 8c3a28f..15d7844 100644
--- a/arch/mips/cavium-octeon/setup.c
+++ b/arch/mips/cavium-octeon/setup.c
@@ -798,22 +798,23 @@ void __init prom_init(void)
 		const char *arg = phys_to_virt(octeon_boot_desc_ptr->argv[i]);
 		if (strncmp(arg, "mem=block:", 10) == 0) {
 			const char *pos = arg + 10;
-			i = 0;
-			for (i = 0; pos[0] && i < ARRAY_SIZE(named_memory_blocks); i++) {
+			int j;
+
+			for (j = 0; pos[0] && j < ARRAY_SIZE(named_memory_blocks); j++) {
 				int len;
 				char *comma = strchr(pos, ',');
 				if (comma)
 					len = comma - pos;
 				else
 					len = max(strlen(pos), ARRAY_SIZE(named_memory_blocks[0]));
-				strncpy(named_memory_blocks[i], pos, len);
+				strncpy(named_memory_blocks[j], pos, len);
 				if (comma)
 					pos = comma + 1;
 				else
 					break;
 			}
-			for (i = 0; i < ARRAY_SIZE(named_memory_blocks); i++)
-				pr_err("Named Block[%d] = \"%s\"\n", i, named_memory_blocks[i]);
+			for (j = 0; j < ARRAY_SIZE(named_memory_blocks); j++)
+				pr_err("Named Block[%d] = \"%s\"\n", j, named_memory_blocks[j]);
 		} else if ((strncmp(arg, "MEM=", 4) == 0) ||
 			   (strncmp(arg, "mem=", 4) == 0)) {
 			sscanf(arg + 4, "%llu", &MAX_MEMORY);
-- 
1.7.5.4

